torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3291,6 +3303,54 @@ def _check_rebuild_full_params(self, param: FlatParameter):
                     )
             eod.param_order.append(param_index)
 
+    def register_comm_hook(self, state: object, hook: callable):
```

Comment:
should we enforce to register hook on the root only?

Reply:
Thanks for pointing that out! Added into the updated version.